src/client/wieldmesh.cpp

```diff
@@ -235,31 +247,41 @@ WieldMeshSceneNode::~WieldMeshSceneNode()
 		g_extrusion_mesh_cache = nullptr;
 }
 
-void WieldMeshSceneNode::setExtruded(const std::string &imagename,
-	const std::string &overlay_name, v3f wield_scale, ITextureSource *tsrc,
-	u8 num_frames)
+void WieldMeshSceneNode::setExtruded(const TileDef &d0, const TileLayer &l0,
+		const TileDef &d1, const TileLayer &l1,
+		v3f wield_scale, ITextureSource *tsrc)
+{
+	setExtruded(extractTexture(d0, l0, tsrc),
+		extractTexture(d1, l1, tsrc), wield_scale);
+	// Add color
+	m_buffer_info.clear();
+	m_buffer_info.emplace_back(l0.has_color, l0.color);
+	m_buffer_info.emplace_back(l1.has_color, l1.color);
+}
+
+void WieldMeshSceneNode::setExtruded(const std::string &image,
+		const std::string &overlay, v3f wield_scale, ITextureSource *tsrc)
+{
+	video::ITexture *texture = tsrc->getTexture(image);
+	video::ITexture *overlay_texture =
+		overlay.empty() ? nullptr : tsrc->getTexture(overlay);
+	setExtruded(texture, overlay_texture, wield_scale);
+}
+
+void WieldMeshSceneNode::setExtruded(video::ITexture *texture,
+		video::ITexture *overlay_texture, v3f wield_scale)
 {
-	video::ITexture *texture = tsrc->getTexture(imagename);
 	if (!texture) {
 		changeToMesh(nullptr);
 		return;
 	}
-	video::ITexture *overlay_texture =
-		overlay_name.empty() ? NULL : tsrc->getTexture(overlay_name);
 
 	core::dimension2d<u32> dim = texture->getSize();
-	// Detect animation texture and pull off top frame instead of using entire thing
-	// FIXME: this is quite unportable, we should be working with the original TileLayer if there's one
-	if (num_frames > 1) {
-		u32 frame_height = dim.Height / num_frames;
-		dim = core::dimension2d<u32>(dim.Width, frame_height);
-	}
```

Comment:
Does this mean that the texture is already correctly split into the `layer.frames` array, such that manual "clipping" is no longer needed?

Reply:
yes